components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/analytics/APIMgtUsageHandler.java

```diff
@@ -39,7 +42,12 @@
 
     public boolean handleRequest(MessageContext mc) {
 
+        TracingSpan responseLatencySpan = (TracingSpan) mc.getProperty(APIMgtGatewayConstants.RESPONSE_LATENCY);
```

Comment:
Where do you check whether tracing is enabled or not? You need to skip these steps if the tracing disabled.

Reply:
Fixed